### PR TITLE
[BACK-50] Get founders of GitHub repository

### PR DESCRIPTION
**Pull request by jonassuess**
### Description of Issue
Fetching the github commit history to return an array of Founders with their names, login names and twitter handles. This method goes trough the commit history of a specific repo and fetches teh first 5 commits, which are most likley the initiators of a project. It then removes duplicates, because several commits can be from the same person, but shouldn't be returned within the Array
### How I implemented
Build a GraphQL query and a method to filter out duplicates
### Other
None

### CODE REVIEW

File: packages/graphql_server/src/api/githubApi.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * Returns a Array of Founders with their names, login names and twitter handles. This method goes trough the commit history of a specific repo
+ * and fetches teh first 5 commits, which are most likley the initiators of a project. It then removes duplicates, because several commits can be from the
+ * same person, but shouldn't be returned within the Array
+ * @param owner: name of the owner of the github repo
+ * @param name: name of the github repo
+ * @returns An Array of the project founders
+ */
+export async function getRepoFounders(owner: string, name: string): Promise<ProjectFounder[]> {
```
Comment (thomas-woehrle) [inline packages/graphql_server/src/api/githubApi.ts:40]: When an invalid values like an empty string is passed an exception is thrown. Please fix this and let it display a suitable message on console.error instead f.e. "Founders of <reponame> owned by <ownername> could not be found"

File: packages/graphql_server/src/api/githubApi.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import axios, { AxiosResponse } from 'axios'
+import dotenv from 'dotenv'
+
+dotenv.config()
+
+type ProjectFounder = {
```
Comment (thomas-woehrle) [inline packages/graphql_server/src/api/githubApi.ts:6]: Refactor types into a separate file in the types folder named `githubApi.types.ts`

**Review by thomas-woehrle — CHANGES_REQUESTED** [review]
Very good job. 
Only minor changes needed.

**Review by thomas-woehrle — APPROVED** [review]
ProjectFounder type could be GitHubUser type as well, but I can change it with the next version of dbUpdater

File: packages/graphql_server/src/api/githubApi.ts [hunk 3]
```diff
@@ -65,3 +71,71 @@ export async function getUserInfo(query: string, authToken: string): Promise<Git
...
+      headers: {
+        Authorization: authToken
+      }
+    }
+  )
+
+  const distinctCommiters: ProjectFounder[] = []
+
+  // checks, whether a login name appears twice and only pushes distinct founders into the array
+  response.data.data.repository.defaultBranchRef.target.history.edges.forEach((node) => {
```
Comment (pczern) [inline packages/graphql_server/src/api/githubApi.ts:129]: response?.data?.data everywhere, also check for existing edges or throw error, also make sure to use optional chaining below

**Review by pczern — CHANGES_REQUESTED** [review]

**Review by kigadye — APPROVED** [review]
Looks good

**Review by pczern — APPROVED** [review]
Please read about optional chaining before merging

**Pull request merged**